### PR TITLE
Template helper

### PR DESCRIPTION
**Pull request by aparlato**
Closes https://github.com/stamen/mapbox-gl-style-build/issues/14

This PR alters existing custom scripts to set up the build system into a general helper script. To run the script in a repo after installing the package, see "Implementation" section in README:
```bash
create-layer-templates
    --in-dir=styles
    --out-dir=templates
    --base-path=styles/base-style.json
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,125 +32,149 @@ mapbox-gl-style-build
...
+```
+
+The parameters are as follows:
+
+- `--in-dir`: the style directory containing existing variant styles to break out into template files
+- `--out-dir`: the directory to build your template files to
+- `--base-path`: the name of the style in the "in-dir" that is the base style
+
+After running the implementation script, you may manually delete the initial styles directory unless reusing it for building the templates to. The script avoids this step as there may occasionally be reason to leave these pre-build-system styles for posterity.
+
```
Comment (aparlato) [inline README.md:164]: This implementation section is all that changed in the README. Everything else is a formatting change from prettier. Let me know if we want to leave this as is, remove the extra formatting, or make a followup issue to add prettier to this repo.

File: src/lib/create-template.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+export const createVariantTemplate = style => {
+  const templateStyle = {
+    ...style,
+    layers: style.layers.map(l => l.id)
+  };
+
+  const fileContent = `module.exports.context = {
+  colors: {
+  },
```
Comment (aparlato) [inline src/lib/create-template.js:99]: Should we remove this `colors` context as too specific?

File: src/lib/create-template.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env node
```
Comment (ebrelsford) [inline src/lib/create-template.js:1]: Do we need this here? I don't think we'll be executing this file directly, right?

File: src/lib/create-template.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        ${allOverrides}
+
+        return {
+            baseStyle,
+            overrides
+        };
+    };`;
+
+  return fileContent;
+};
```
Comment (ebrelsford) [inline src/lib/create-template.js:89]: Can we reduce the indentation here a bit? Here's what I'm seeing:

```js
module.exports.default = (context) => {
        const baseStyle = {
    "id": "layer1",
    "type": "symbol",
    "source": "test",
    "source-layer": "test",
    "minzoom": 14.5,
    "filter": [
      "all",
      ["==", ["geometry-type"], "Point"],
    ],
    "layout": {
      "icon-anchor": "bottom",
      "icon-image": "",
      "icon-rotation-alignment": "viewport",
      "text-anchor": "top",
      "text-field": ["coalesce", ["get", "name:en"], ["get", "name"]],
      "text-letter-spacing": 0.05,
      "text-line-height": 1,
      "text-offset": [0, 0.2],
      "text-size": ["interpolate", ["linear"], ["zoom"], 14, 10, 16, 11]
    },
    "paint": {
      "text-color": "hsl(120, 1%, 65%)",
      "text-halo-blur": 0.5,
      "text-halo-color": "hsl(97, 16%, 95%)",
      "text-halo-width": 1.25,
      "text-opacity": ["step", ["zoom"], 0, 15.5, 1]
    }
  };

        let overrides = {};

        if (context.styleName === 'style1') {
        overrides = {};
    }

        return {
            baseStyle,
            overrides
        };
    };
```

Maybe something like

```suggestion
  const fileContent = `module.exports.default = (context) => {
  const baseStyle = ${baseStyle};

  let overrides = {};

  ${allOverrides}

  return {
    baseStyle,
    overrides
  };
};`;

  return fileContent;
};
```

File: src/lib/create-template.js (hunk 2 above)
Comment (ebrelsford) [reply src/lib/create-template.js:99]: I'm okay with leaving it.

File: bin/create-layer-templates [hunk 5]
```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
+const yargs = require('yargs/yargs');
+const { hideBin } = require('yargs/helpers');
+const { createLayerTemplate, createVariantTemplate } = require('../dist/main');
+
+const argv = yargs(hideBin(process.argv))
```
Comment (ebrelsford) [inline bin/create-layer-templates:8]: Can we add some basic validation where we try opening each of the directories/files and output a message if any fails?

File: bin/create-layer-templates [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    describe: 'The input style directory',
+    type: 'string'
+  })
+  .option('out-dir', {
+    alias: 'outDir',
+    default: 'templates',
+    describe: 'The output directory',
+    type: 'string'
+  })
+  .option('base-path', {
```
Comment (ebrelsford) [inline bin/create-layer-templates:22]: Can we call this `base-style-path` instead? `base-path` sounds like all operations are going to start from that path, which isn't the case.

**Review by ebrelsford — CHANGES_REQUESTED** [review]
Sorry for the delay. This generally looks great! I have a few small tweaks I'd appreciate having in place when you can.

**Comment by aparlato** [comment]
Thanks @ebrelsford ! Just made updates, back to you!

**Review by ebrelsford — APPROVED** [review]
Thanks!

**Pull request merged**